### PR TITLE
Update deps to support project upgrades feature

### PR DESCRIPTION
**Pull request by stwiname**
# Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have tested locally
- [ ] I have performed a self review of my changes
- [x] Updated any relevant documentation
- [x] Linked to any relevant issues
- [x] I have added tests relevant to my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] My code is up to date with the base branch
- [ ] I have updated relevant changelogs. [We suggest using chan](https://github.com/geut/chan/tree/main/packages/chan)

### CODE REVIEW

File: packages/node/package.json [hunk 1]
```diff
@@ -27,9 +27,10 @@
     "@nestjs/event-emitter": "^2.0.0",
     "@nestjs/platform-express": "^9.4.0",
     "@nestjs/schedule": "^3.0.1",
-    "@subql/common": "2.6.0",
+    "@subql/common": "2.7.1-3",
     "@subql/common-cosmos": "workspace:*",
-    "@subql/node-core": "^4.0.1",
+    "@subql/node-core": "4.2.4-26",
+    "@subql/types-core": "0.0.0-2",
```
Comment (stwiname) [inline packages/node/package.json:33]:
```suggestion
```

File: packages/node/src/configure/configure.module.ts [hunk 2]
```diff
@@ -1,157 +1,57 @@
...
-    if (['_', '$0'].includes(key)) return acc;
-
-    if (key === 'network-registry') {
-      try {
-        value = JSON.parse(value as string);
-      } catch (e) {
-        throw new Error('Argument `network-registry` is not valid JSON');
-      }
-    }
-    if (key === 'network-endpoint-param') {
```
Comment (stwiname) [inline packages/node/src/configure/configure.module.ts:43]: Are we losing functionality here? I don't recall what these do?

File: packages/node/src/subcommands/reindex.module.ts [hunk 3]
```diff
@@ -17,18 +16,21 @@ import { ApiService } from '../indexer/api.service';
 import { DsProcessorService } from '../indexer/ds-processor.service';
 import { DynamicDsService } from '../indexer/dynamic-ds.service';
 import { UnfinalizedBlocksService } from '../indexer/unfinalizedBlocks.service';
-import { ReindexService } from './reindex.service';
 
 @Module({
   providers: [
     StoreCacheService,
     StoreService,
     ReindexService,
```
Comment (stwiname) [inline packages/node/src/subcommands/reindex.module.ts:24]: Missing POIService. https://github.com/subquery/subql/commit/a60f85d6c2c16c74ea349d30710616cae097d7fb#diff-7006064ee2e455544671a4ce906c2f2b8dd794a20a41e0a2db5e9a5387fc3c9a

File: packages/types/src/project.ts [hunk 4]
```diff
@@ -3,11 +3,23 @@
 
 import {CosmWasmClient} from '@cosmjs/cosmwasm-stargate';
 import {Registry} from '@cosmjs/proto-signing';
+import {
```
Comment (stwiname) [inline packages/types/src/project.ts:6]: This file is missing the JSDoc improvements

**Review by stwiname — COMMENTED** [review]
This commit needs to be applied https://github.com/subquery/subql/commit/296a8fa0eef30612ef676066381c71d9ceb63661

**Review by jiqiang90 — CHANGES_REQUESTED** [review]

**Pull request merged**